Content.Client/SubFloor/TrayScannerSystem.cs

```diff
@@ -0,0 +1,134 @@
+using System.Collections.Generic;
+using System.Linq;
+using Content.Shared.SubFloor;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Maths;
+
+namespace Content.Client.SubFloor;
+
+public class TrayScannerSystem : SharedTrayScannerSystem
+{
+    [Dependency] private IEntityLookup _entityLookup = default!;
+    [Dependency] private SubFloorHideSystem _subfloorSystem = default!;
+    [Dependency] private SharedContainerSystem _containerSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TrayScannerComponent, TrayScannerToggleEvent>(OnTrayScannerToggle);
+    }
+
+    private void OnTrayScannerToggle(EntityUid uid, TrayScannerComponent scanner, TrayScannerToggleEvent args)
+    {
+        UpdateTrayScanner(uid, scanner);
+
+        if (scanner.Toggled) _activeScanners.Add(uid);
+    }
+
+    private HashSet<EntityUid> _activeScanners = new();
+    private HashSet<EntityUid> _invalidScanners = new();
+
+    public override void Update(float frameTime)
+    {
+        if (!_activeScanners.Any()) return;
+
+        foreach (var scanner in _activeScanners)
+        {
+            if (!UpdateTrayScanner(scanner))
+                _invalidScanners.Add(scanner);
+        }
+
+        _activeScanners.ExceptWith(_invalidScanners);
+        _invalidScanners.Clear();
```

Comment:
is the idea here to remove any invalid scanners from activescanners? if so, use a RemQueue\<T\>

Reply:
yeah, that's what this is supposed to do - i guess i'll try the RemQueue and see if that's more satisfactory